stdlib/universe/fill.go

```diff
@@ -0,0 +1,268 @@
+	plan.RegisterProcedureSpec(FillKind, newFillProcedure, FillKind)
+	execute.RegisterTransformation(FillKind, createFillTransformation)
+}
+
+func createFillOpSpec(args flux.Arguments, a *flux.Administration) (flux.OperationSpec, error) {
+	if err := a.AddParentFromArgs(args); err != nil {
+		return nil, err
+	}
+
+	spec := new(FillOpSpec)
+
+	if col, err := args.GetRequiredString("column"); err != nil {
+		return nil, err
+	} else {
+		spec.Column = col
+	}
+
+	if usePrevious, ok, err := args.GetBool("usePrevious"); err != nil {
+		return nil, err
+	} else if ok {
+		spec.UsePrevious = usePrevious
+	}
+
+	if val, err := args.GetRequired("value"); err != nil {
+		return nil, err
+	} else {
+		typ := val.Type()
+		spec.Type = typ.Nature().String()
+		switch typ {
+		case semantic.Bool:
+			spec.Value = strconv.FormatBool(val.Bool())
+		case semantic.Int:
+			spec.Value = strconv.FormatInt(val.Int(), 10)
+		case semantic.UInt:
+			spec.Value = strconv.FormatUint(val.UInt(), 10)
+		case semantic.Float:
+			spec.Value = strconv.FormatFloat(val.Float(), 'f', -1, 64)
+		case semantic.String:
+			spec.Value = val.Str()
+		case semantic.Time:
+			spec.Value = val.Time().String()
+		default:
+			return nil, errors.New("value type for fill must be a valid primitive type (bool, int, uint, float, string, time)")
+		}
+
+	}
+
+	return spec, nil
+}
+
+func newFillOp() flux.OperationSpec {
+	return new(FillOpSpec)
+}
+
+func (s *FillOpSpec) Kind() flux.OperationKind {
+	return FillKind
+}
+
+type FillProcedureSpec struct {
+	plan.DefaultCost
+	Column      string
+	Value       values.Value
+	UsePrevious bool
+}
+
+func newFillProcedure(qs flux.OperationSpec, pa plan.Administration) (plan.ProcedureSpec, error) {
+	spec, ok := qs.(*FillOpSpec)
+	if !ok {
+		return nil, fmt.Errorf("invalid spec type %T", qs)
+	}
+
+	pspec := &FillProcedureSpec{
+		Column:      spec.Column,
+		UsePrevious: spec.UsePrevious,
+	}
+
+	switch spec.Type {
+	case "bool":
+		v, err := strconv.ParseBool(spec.Value)
+		if err != nil {
+			return nil, err
+		}
+		pspec.Value = values.New(v)
+	case "int":
+		v, err := strconv.ParseInt(spec.Value, 10, 64)
+		if err != nil {
+			return nil, err
+		}
+		pspec.Value = values.New(v)
+	case "uint":
+		v, err := strconv.ParseUint(spec.Value, 10, 64)
+		if err != nil {
+			return nil, err
+		}
+		pspec.Value = values.New(v)
+	case "float":
+		v, err := strconv.ParseFloat(spec.Value, 64)
+		if err != nil {
+			return nil, err
+		}
+		pspec.Value = values.New(v)
+	case "string":
+		pspec.Value = values.New(spec.Value)
+	case "time":
+		v, err := values.ParseTime(spec.Value)
+		if err != nil {
+			return nil, err
+		}
+		pspec.Value = values.New(v)
+	default:
+		return nil, errors.New("unknown type in fill op-spec")
+	}
+
+	return pspec, nil
+}
+
+func (s *FillProcedureSpec) Kind() plan.ProcedureKind {
+	return FillKind
+}
+func (s *FillProcedureSpec) Copy() plan.ProcedureSpec {
+	ns := new(FillProcedureSpec)
+
+	*ns = *s
+	ns.Value = s.Value
```

Comment:
This is my ignorance 😅 
You don't need to copy primitiva values because they are copied at the line before when assigning the content of objects?

Reply:
yeah, and probably the value is OK too since they are only primitives but I though to just be safe.

Reply:
I agree this copy isn't needed. `values.Value` is supposed to be an immutable interface so, in the case that the pointer gets copied, it should still be fine since the value itself is immutable.